src/commands/mod.rs

```diff
@@ -2005,14 +2008,71 @@ fn cmd_new(ui: &mut Ui, command: &CommandHelper, args: &NewArgs) -> Result<(), C
         "expected a non-empty list from clap"
     );
     let commits = resolve_base_revs(&workspace_command, &args.revisions)?;
-    let parent_ids = commits.iter().map(|c| c.id().clone()).collect();
+    let parent_ids = commits.iter().map(|c| c.id().clone()).collect::<Vec<_>>();
     let mut tx = workspace_command.start_transaction("new empty commit");
-    let merged_tree = merge_commit_trees(tx.base_repo().as_repo_ref(), &commits);
-    let new_commit = tx
-        .mut_repo()
-        .new_commit(command.settings(), parent_ids, merged_tree.id().clone())
-        .set_description(&args.message)
-        .write()?;
+    let mut num_rebased = 0;
+    let new_commit;
+    if args.insert_before {
+        // Instead of having the new commit as a child of the changes given on the
+        // command line, add it between the changes parents and the changes.
```

Comment:
Needs an apostrophe: "...between the changes' parents and the changes themselves"

Reply:
Done.